### PR TITLE
feat: Add JWT token type

### PR DESCRIPTION
**Pull request by pi-kei**
Fix: https://github.com/casdoor/casdoor-go-sdk/issues/99#issuecomment-1954464072

Add `TokenType` field to `casdoorsdk.Claims`.
Add `IsRefreshToken` func to `casdoorsdk.Claims`.
This feature will let you find out if you are dealing with refresh token or access token.

### CODE REVIEW

File: casdoorsdk/jwt.go
Comment (hsluoyz) [inline casdoorsdk/jwt.go:27]: Did you test your code? Shouldn't it be `json:"tokenType"`?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: casdoorsdk/jwt.go
Comment (pi-kei) [reply casdoorsdk/jwt.go:27]: There is `tokenType` which equals to `access-token` in both refresh and access tokens. But `TokenType` which equals to `refresh-token` presents only in refresh tokens. Not sure why. Maybe it is because of misspell in server code.

**Review by hsluoyz — APPROVED** [review]

**Comment by pi-kei** [comment]
@hsluoyz I'm sorry but I've not tested this properly. So, it is not working as expected. Now `casdoorsdk.Claims.TokenType` is always equals to `access-token` because of how json decoder works.

Here's the quote from [docs](https://pkg.go.dev/encoding/json#Unmarshal):

> To unmarshal JSON into a struct, Unmarshal matches incoming object keys to the keys used by Marshal (either the struct field name or its tag), preferring an exact match but also accepting a case-insensitive match.

Right now it's treating `tokenType` and `TokenType` as the same field. To fix that there must be both of them in struct. I can do a fix PR for that.

**Comment by hsluoyz** [comment]
@pi-kei I told you to test your code locally before submitting it. Now this PR will be reverted. And I have to see your demo video to show the test case running well (but don't submit it) before merging your PR again

**Pull request merged**
